x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -200,6 +216,41 @@ export function XYChart({ data, args, formatFactory, timeZone }: XYChartRenderPr
               }
             : undefined
         }
+        onElementClick={([[geometry, series]]) => {
+          const layer = layers.find(l =>
+            series.seriesKeys.some(key => l.accessors.includes(key as string))
+          )!;
+          const table = data.tables[layer.layerId];
+
+          const points = [
+            {
+              row: table.rows.findIndex(row => row[layer.xAccessor] === geometry.x),
+              column: table.columns.findIndex(col => col.id === layer.xAccessor),
+              value: geometry.x,
+            },
+          ];
+
+          if (series.seriesKeys.length > 1) {
+            const pointValue = series.seriesKeys[0];
+
+            points.push({
+              row: table.rows.findIndex(row => row[layer.splitAccessor] === pointValue),
+              column: table.columns.findIndex(col => col.id === layer.splitAccessor),
+              value: pointValue,
+            });
+          }
```

Comment:
@markov00 This logic is the logic I was expecting to have included in the chart library itself